File: Enrollify/src/Disciplina.java
import java.util.List;

public class Disciplina {
    public int id;
    public String nome;
    public int creditos;
    public int vagasTotais;
    public int vagasOcupadas;
    public List<Aluno> alunosMatriculados;
    public Professor professor;

    public boolean adicionarAluno(Aluno aluno) {
        // TODO
        return false;
    }

    public void removerAluno(Aluno aluno) {
        // TODO
    }

    public boolean verificarDisponibilidade() {
        // TODO
        return false;
    }
}

File: Enrollify/VersaoJavaScript/Disciplina.js
class Disciplina {
    static MIN_ALUNOS = 3;
    static MAX_ALUNOS = 60;
    
    constructor(id, nome, credito, valor, professor) {
        this.id = id;
        this.nome = nome;
        this.credito = credito;
        this.valor = valor;
        this.alunosMatriculados = [];
        this.professor = professor;
        this.status = EstadoDisciplina.ATIVA;
    }
    
    adicionarAluno(aluno) {
        if (this.alunosMatriculados.length < Disciplina.MAX_ALUNOS) {
            this.alunosMatriculados.push(aluno);
            return true;
        }
        return false;
    }
    
    removerAluno(aluno) {
        this.alunosMatriculados = this.alunosMatriculados.filter(a => a.id !== aluno.id);
        if (this.alunosMatriculados.length < Disciplina.MIN_ALUNOS) {
            this.status = EstadoDisciplina.CANCELADA;
        }
    }
}